src/main/java/io/javalin/Javalin.java

```diff
@@ -732,4 +734,34 @@ void clearMatcherAndMappers() {
         exceptionMapper.getExceptionMap().clear();
     }
 
+    // Extension
+
+    /**
+     * Registers an {@link Extension} with the Javalin application.
+     *
+     * @param extension You're free to implement the extension as a class or a lambda expression.
+     * @return Self instance for fluent, method-chaining API
+     */
+    public Javalin extension(Extension extension) {
+        extension.register(this);
+
+        String clazzName = extension.getClass().getCanonicalName();
+        if (clazzName != null) {
```

Comment:
@dherges what do you think about making it explicit, so that you only have a getter if you give class as first arg:

```
app.extension(MyExtensions.class, new MyExtension()); // has getter: app.extension(MyExtensions.class);

a.extension(new MyExtension()); // no getter
a.extension(app -> {}); // no getter
```